AudioStream.js

```diff
@@ -97,29 +98,44 @@ module.exports = class AudioStream {
         }
     }
 
-
-    async extractVideoIds(url) {
+    async extractVideoIdsFromPlaylist(playlistUrl){
+        
         const s = new Set();
-        try {
-            let temp = await fetch(url).then(res => res.text());
-            while (temp !== "") {
-                const ind = temp.indexOf('videoId');
-                if (temp.substring(ind, ind + 40).includes("videoId")) {
-                    const videoIdStart = ind + 10;
-                    const videoIdEnd = ind + 23;
-                    const videoId = temp.substring(videoIdStart, videoIdEnd).replace(/["\[\],]/g, "");
-                    console.log(temp.substring(ind, ind + 40))
-                    s.add("https://www.youtube.com/watch?v=" + videoId);
-                }
-                temp = temp.substring(ind + 21);
-            }
-        } catch (error) {
-            console.error('Error:', error);
-        }
-        for (const video of Array.from(s)) {
-            this.queueUp(video)
-        };
-    }
+        const urlParams = new URLSearchParams(new URL(playlistUrl).search);
```

Comment:
i think you can just do this and it will work the same

```suggestion
        const urlParams = new URLSearchParams(playlistUrl);
```

Reply:
The URLSearchParams only takes in a query string I think, so I have to extract it first.